snowplow_tracker/events/self_describing.py

```diff
@@ -0,0 +1,91 @@
+from snowplow_tracker.events.event import Event
+from snowplow_tracker import SelfDescribingJson
+
+# """
+#     self_describing.py
+
+#     Copyright (c) 2013-2023 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+# """
+from typing import Optional
+from snowplow_tracker.typing import JsonEncoderFunction
+
+from snowplow_tracker.constants import UNSTRUCT_EVENT_SCHEMA
+
+
+class SelfDescribing(Event):
```

Comment:
Can you copy the API docs for these classes from the Java tracker? https://github.com/snowplow/snowplow-java-tracker/blob/master/src/main/java/com/snowplowanalytics/snowplow/tracker/events/SelfDescribing.java#L24-L29